src/braket/experimental/autoqasm/program/program.py

```diff
@@ -404,14 +423,31 @@ def get_expression_var(self, expression: FreeParameterExpression) -> oqpy.FloatV
         self.get_oqpy_program().declare(var)
         return var
 
-    def get_free_parameters(self) -> list[oqpy.FloatVar]:
+    def get_free_parameters(self) -> list[oqpy.Var]:
         """Return a list of named oqpy.Vars that are used as free parameters in the program."""
         return list(self._free_parameters.values())
 
+    def get_free_parameter(self, name: str) -> oqpy.Var | None:
+        """Return the oqpy.Var associated with the variable name `name` in the program."""
+        return self._free_parameters.get(name, None)
```

Comment:
Consider adjusting naming for consistency:
- `_free_parameters` --> `_input_parameters`
- `get_free_parameters` --> `_get_input_parameters`
- `get_free_parameter` --> `_get_input_parameter`